src/core/types/addons.js

```diff
@@ -61,6 +61,8 @@ export type ExternalAddonType = {|
   authors?: Array<AddonAuthorType>,
   average_daily_users?: number,
   categories?: Object,
+  click_data?: string | null,
+  click_url?: string | null,
```

Comment:
Shouldn't we have a type similar to `CollectionAddonType` instead?

Reply:
See https://github.com/mozilla/addons-server/pull/15670#issuecomment-705219144. This is the "cheat" I was talking about in the following comment. Ideally we should have a unique type for this, but it really complicates the code, both in terms of types and in terms of how we build an internal add-on. Because we have stated that increasing our development time as a result of typing issues is a "bad thing", I chose to just implement this simple solution. It's definitely not perfect, if we're looking for perfect typing, but it also doesn't really break anything.

Reply:
In case it isn't obvious from the comment I linked to, the reason we can have a `CollectionAddonType` without encountering these issues is that the extra fields exist outside of the `addon` in the API response, but I don't think that really makes sense from an API design perspective.

Reply:
I see, thanks for the explanation. Maybe add some comments to explain where these props come from and why.

Reply:
Wait. I am confused actually. Re-reading the code in this patch, it looks like the impression URLs are not bundled with each add-on but rather returned once for all the add-ons, so I don't understand.

Reply:
Ha, I mixed impression and click URLs for some reasons. Ok, I see why it's like that now and that probably make sense, although I don't understand why we cannot have a new type for the add-on returned by this sponsored API endpoint. The saga does not load the `addons` in the `addons` reducer.

Reply:
It's true, but I wanted to reuse `createInternalAddon` from the `addons` reducer. Without this method I either need to copy the code for that into the `shelves` reducer, or I need to do a bunch of complicated stuff with types. I tried some simple approaches with types first, but none of them worked. I'm just trying to avoid duplicating the code for `createInternalAddon`, or adding hacky code like removing the properties, calling `createInternalAddon`, and then adding the properties back in.

Reply:
Ha! I see. Sorry, I am very slow today. Okay well, I should debug that on my own but I think what we have here is very acceptable so it's fine by me. We can still revisit later if needed. Let's have some comments.